storage/store.go

```diff
@@ -315,6 +315,9 @@ type StoreContext struct {
 
 	// EventFeed is a feed to which this store will publish events.
 	EventFeed *util.Feed
+
+	// Tracer is a request tracer used by this store for request tracing.
+	Tracer *util.Tracer
```

Comment:
I'll play around with the API. Probably a Trace object which already knows its TraceID as you suggest above is a good candidate for putting into the context.